src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -532,15 +537,13 @@ public ConnectionString Resolve(bool resolveHosts)
 
             var host = GetHostNameForDns();
 
-            var client = new LookupClient();
-
             ConnectionStringScheme resolvedScheme;
             List<string> hosts;
             if (resolveHosts)
             {
                 resolvedScheme = ConnectionStringScheme.MongoDB;
-                var srvResponse = client.Query(srvPrefix + host, QueryType.SRV, QueryClass.IN);
-                hosts = GetHostsFromResponse(srvResponse);
+                var srvRecords = _dnsResolver.ResolveSrvRecords(srvPrefix + host, CancellationToken.None);
```

Comment:
should this method (line 531) also have CancelaltionToken?

Reply:
That sounds like a good idea.

You might have to add a CancellationToken to the methods that call this method also.

It's possible that some code paths that end up here won't have any way of providing a CancellationToken.

Reply:
done